chain-abci/src/enclave_bridge/edp/tdbe.rs

```diff
@@ -42,58 +42,62 @@ pub struct TdbeApp {
         tdbe_config: &TdbeConfig,
         ra_config: &SpRaConfig,
-        storage: Arc<dyn KeyValueDB>,
+        _storage: Arc<dyn KeyValueDB>,
+        tve_stream: UnixStream,
     ) -> std::io::Result<Self> {
         // - `chain_abci_stream` is passed to enclave. Encalve can send requests to chain-abci
         //   using this
         // - `chain_abci_receiver` listens to the requests sent by enclave and responds to them
-        let (chain_abci_stream, chain_abci_receiver) = UnixStream::pair()?;
+        let (chain_abci_stream, _chain_abci_receiver) = UnixStream::pair()?;
 
         // - `persistence_stream` is passed to enclave. Encalve can send requests to chain-storage
         //   using this
         // - `persistence_receiver` listens to the requests sent by enclave and responds to them
-        let (persistence_stream, persistence_receiver) = UnixStream::pair()?;
+        let (persistence_stream, _persistence_receiver) = UnixStream::pair()?;
 
-        spawn_chain_abci_thread(chain_abci_receiver, storage.clone());
-        spawn_persistence_thread(persistence_receiver, storage);
+        // FIXME: spawn these when they actually do something
+        // spawn_chain_abci_thread(chain_abci_receiver, storage.clone());
+        // spawn_persistence_thread(persistence_receiver, storage);
 
         Ok(Self {
             chain_abci_stream,
             persistence_stream,
             sp_address: ra_config.address.clone(),
             remote_rpc_address: tdbe_config.remote_rpc_address.clone(),
             local_listen_address: tdbe_config.local_listen_address.clone(),
+            tve_stream,
         })
     }
 
     pub fn spawn(self) {
-        thread::spawn(move || {
-            let mut device = Device::new()
-                .expect("SGX device was not found")
-                .einittoken_provider(AesmClient::new())
-                .build();
-            let mut enclave_builder = EnclaveBuilder::new("tdb-enclave-app.sgxs".as_ref());
-
-            enclave_builder
-                .coresident_signature()
-                .expect("Enclave signature file not found");
-            enclave_builder.usercall_extension(self);
-
-            let enclave = enclave_builder
-                .build(&mut device)
-                .expect("Failed to build enclave");
-            enclave.run().expect("Failed to start enclave")
-        });
+        let mut device = Device::new()
+            .expect("SGX device was not found")
+            .einittoken_provider(AesmClient::new())
+            .build();
+        let mut enclave_builder = EnclaveBuilder::new("tdb-enclave-app.sgxs".as_ref());
+
+        enclave_builder
+            .coresident_signature()
+            .expect("Enclave signature file not found");
+        enclave_builder.usercall_extension(self);
+
+        let enclave = enclave_builder
+            .build(&mut device)
+            .expect("Failed to build enclave");
+        enclave.run().expect("Failed to start enclave")
```

Comment:
This call will block the main thread. Maybe call `enclave.run()` in a separate thread?

Reply:
as per yesterday's discussion -- it doesn't need to do anything else in the meantime and the rest of the application can't continue

Reply:
We can only run one thing in the current thread. If we block for enclave, we'll have to spawn different threads for persistence and chain-abci? I thought you meant to run persistence code on the main thread and spawn threads for other things?

Reply:
oh, yes, it was the persistence on main (instead of `spawn_persistence_thread`) -- but it may actually be required for storing the trusted anchors + sealed keypackage secrets.

For this one, it should be spawned on a separate thread -- but at this moment, I put it on main, because it needs to signal back that it completed (or failed) ...

There are several missing things at the moment compared to what it should signal in the future: https://github.com/crypto-com/chain-docs/blob/master/docs/modules/tdbe.md#newrejoining-node-process
1. after catching up and completing persistence of sealed transactions payloads, it'll also need to send back to chain-abci the `Add`+ `Commit` payloads for node join construction
2. it'll need to send back the trusted anchors + sealed keypackage secrets once the nodejoin is accepted

until 2., chain-abci main thread needs to block -- how is it going to be signalled? via the persistence thread?

Reply:
it's on a separate thread now + there's an explanation